ReactiveCocoa/Swift/EventLogger.swift

```diff
@@ -0,0 +1,88 @@
+	
+	public static let allEvents: Set<SignalProducerLoggingEvent> = {
+		return [.Started, .Next, .Completed, .Failed, .Terminated, .Disposed, .Interrupted]
+	}()
+}
+
+private func printLog(identifier: String, event: String, fileName: String, functionName: String, lineNumber: Int) -> Void {
+	print("[\(identifier)] \(event) fileName: \(fileName), functionaName: \(functionName), lineNumber: \(lineNumber)")
+}
+
+public typealias EventLogger = (identifier: String, event: String, fileName: String, functionName: String, lineNumber: Int) -> Void
+
+extension SignalType {
+	/// Logs all events that the receiver sends.
+	/// By default, it will print to the standard output.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func logEvents(identifier: String = "", events: Set<SignalLoggingEvent> = SignalLoggingEvent.allEvents, fileName: String = #file, functionName: String = #function, lineNumber: Int = #line, logger: EventLogger = printLog) -> Signal<Value, Error> {
+		
+		let logEvent: String -> Void = { event in
+			logger(identifier: identifier, event: event, fileName: fileName, functionName: functionName, lineNumber: lineNumber)
+		}
+		
+		func log(event: SignalLoggingEvent) -> (Void -> Void)? {
+			return events.contains(event) ? { logEvent("\(event.rawValue)") } : nil
+		}
+		
+		func logValue<T>(event: SignalLoggingEvent) -> (T -> Void)? {
+			return events.contains(event) ? { logEvent("\(event.rawValue) \($0)") } : nil
+		}
+		
+		return self.on(
+			failed: logValue(.Failed),
+			completed: log(.Completed),
+			interrupted: log(.Interrupted),
+			terminated: log(.Terminated),
+			disposed: log(.Disposed),
+			next: logValue(.Next))
+	}
+}
+
+extension SignalProducerType {
+	/// Logs all events that the receiver sends.
+	/// By default, it will print to the standard output.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func logEvents(identifier: String = "", events: Set<SignalProducerLoggingEvent> = SignalProducerLoggingEvent.allEvents, fileName: String = #file, functionName: String = #function, lineNumber: Int = #line, logger: EventLogger = printLog) -> SignalProducer<Value, Error> {
+		
+		let logEvent: String -> Void = { event in
+			logger(identifier: identifier, event: event, fileName: fileName, functionName: functionName, lineNumber: lineNumber)
+		}
+		
+		func log(event: SignalProducerLoggingEvent) -> (Void -> Void)? {
+			return events.contains(event) ? { logEvent("\(event.rawValue)") } : nil
+		}
+		
+		func logValue<T>(event: SignalProducerLoggingEvent) -> (T -> Void)? {
+			return events.contains(event) ? { logEvent("\(event.rawValue) \($0)") } : nil
+		}
+		
+		return self.on(
+			started: log(.Started),
+			failed: logValue(.Failed),
+			completed: log(.Completed),
+			interrupted: log(.Interrupted),
+			terminated: log(.Terminated),
+			disposed: log(.Disposed),
+			next: logValue(.Next))
```

Comment:
It's rendering correctly on my side. 😕 
<img width="311" alt="screen shot 2016-05-02 at 12 21 49" src="https://cloud.githubusercontent.com/assets/949286/14953312/7b8e118c-1060-11e6-9c41-e5f8087b6bf0.png">